### PR TITLE
JsonRequestLog: reshuffle to allow replacing message field

### PR DESCRIPTION
**Pull request by stevenschlansker**

### CODE REVIEW

**Comment by pennello** [comment]
Taking a look.

File: jetty/src/main/java/com/opentable/logging/jetty/JsonRequestLog.java [hunk 1]
```diff
@@ -143,6 +143,23 @@ protected HttpV1 createEvent(Request request, Response response) {
             .build();
     }
 
+    protected String constructMessage(HttpV1 payload) {
+        return constructMessage0(payload);
```
Comment (mikebell90) [inline jetty/src/main/java/com/opentable/logging/jetty/JsonRequestLog.java]: why not just defaultMessageConstruction

File: jetty/src/main/java/com/opentable/logging/jetty/JsonRequestLog.java [hunk 2]
```diff
@@ -81,7 +81,7 @@ public void log(final Request request, final Response response)
         }
 
         final HttpV1 payload = createEvent(request, response);
```
Comment (pennello) [inline jetty/src/main/java/com/opentable/logging/jetty/JsonRequestLog.java]: Is `createEvent` a function the programmer overrides?  If so, consider adding a `@NonNull` annotation.  That'll get included if you override, right?  That might help detect inappropriately returning `null` a little earlier, avoiding the runtime error.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,11 @@
 otj-logging
 ===========
 
+2.7.1
```
Comment (pennello) [inline CHANGELOG.md]: You might want to bump the minor since the public API changed.

**Pull request merged**